igvc_navigation/src/path_follower/main.cpp

```diff
@@ -66,7 +66,10 @@ void position_callback(const nav_msgs::OdometryConstPtr& msg)
   RobotState cur_pos(msg);
 
   double goal_dist = cur_pos.distTo(waypoint->point.x, waypoint->point.y);
-  ROS_INFO_STREAM("Distance to waypoint: " << goal_dist << "(m.)");
+  if (debug)
+  {
+      ROS_INFO_STREAM("Distance to waypoint: " << goal_dist << "(m.)");
```

Comment:
If this information is only used for debug, consider using [ROS_DEBUG_STREAM](http://wiki.ros.org/roscpp/Overview/Logging). Using the built in log levels will allow you to turn debug information on and off without recompiling your code.

Reply:
I use the `debug` conditional to prevent the message from being output to stdout when output="screen" for the ros node. Seeing as ROS_DEBUG_STREAM still outputs to stdout, it might be worthwhile for me to leave it like this.

Reply:
The verbosity level of ROS can be set outside of the application such that DEBUG output does not print to the terminal.

Reply:
Oh that's really helpful. I'll switch over to this then.